crates/aptos/src/update/prover_dependency.rs

```diff
@@ -0,0 +1,363 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    cli_build_information,
+    common::types::{CliCommand, CliError, CliTypedResult},
+    update::{get_additional_binaries_dir, update_helper::get_arch},
+};
+use anyhow::{Context, Result};
+use aptos_build_info::BUILD_OS;
+use async_trait::async_trait;
+use clap::Parser;
+use dirs::home_dir;
+use futures::StreamExt;
+use move_prover_boogie_backend::options::{
+    BoogieOptions, MAX_BOOGIE_VERSION, MAX_CVC5_VERSION, MAX_Z3_VERSION, MIN_BOOGIE_VERSION,
+    MIN_CVC5_VERSION, MIN_Z3_VERSION,
+};
+use reqwest::Client;
+#[cfg(unix)]
+use std::os::unix::fs::PermissionsExt;
+use std::{
+    env,
+    fs::{OpenOptions, Permissions},
+    io::Write,
+    path::{Path, PathBuf},
+    process::Command,
+};
+use tokio::fs::{self};
+
+const GITHUB_URL: &str = "https://github.com/";
+const REPO_NAME: &str = "prover-dependency";
+const REPO_OWNER: &str = "aptos-labs";
+
+const BOOGIE_BINARY_NAME: &str = "boogie";
+const TARGET_BOOGIE_VERSION: &str = "3.2.4.0";
+
+const BOOGIE_EXE_ENV: &str = "BOOGIE_EXE";
+#[cfg(target_os = "windows")]
+const BOOGIE_EXE: &str = "boogie.exe";
+#[cfg(not(target_os = "windows"))]
+const BOOGIE_EXE: &str = "boogie";
+
+const Z3_BINARY_NAME: &str = "z3";
+const TARGET_Z3_VERSION: &str = "4.11.2";
+
+const Z3_EXE_ENV: &str = "Z3_EXE";
+#[cfg(target_os = "windows")]
+const Z3_EXE: &str = "z3.exe";
+#[cfg(not(target_os = "windows"))]
+const Z3_EXE: &str = "z3";
+
+const CVC5_BINARY_NAME: &str = "cvc5";
+const TARGET_CVC5_VERSION: &str = "0.0.3";
+
+const CVC5_EXE_ENV: &str = "CVC5_EXE";
+#[cfg(target_os = "windows")]
+const CVC5_EXE: &str = "cvc5.exe";
+#[cfg(not(target_os = "windows"))]
+const CVC5_EXE: &str = "cvc5";
+
+/// Install dependencies (boogie, z3 and cvc5) for Move prover
+#[derive(Debug, Parser)]
+pub struct ProverDependencyInstaller {
+    /// Where to install binaries of boogie, z3 and cvc5. If not
+    /// given we will put it in a standard location for your OS.
+    #[clap(long)]
+    install_dir: Option<PathBuf>,
+}
+
+impl ProverDependencyInstaller {
+    fn add_env_var(&self, env_var: &str, install_path: &Path) -> Result<(), CliError> {
+        if let Ok(current_value) = env::var(env_var) {
+            if current_value == install_path.to_string_lossy() {
+                return Ok(());
+            } else {
+                println!(
+                    "{} is already set to a different value: {}, please set it manually.",
+                    env_var, current_value
+                );
+                return Ok(());
+            }
+        }
+
+        if cfg!(target_os = "windows") {
+            self.set_env_on_windows(env_var, install_path)?;
+        } else {
+            self.set_env_on_unix(env_var, install_path)?;
+        }
+        Ok(())
+    }
+
+    fn set_env_on_unix(&self, env_var: &str, install_path: &Path) -> Result<(), CliError> {
```

Comment:
None of the other commands set the env var permanently, we leave it to the user. I think trying this is fraught with peril, different envs do different things, there are heaps of different files you would have to think about: `.bashrc`, `.bash_profile`, `.profile`, `.zshrc`, etc. I'm not sure we can assume the user has a `.profile` file.

I see this doesn't cause the whole thing to fail, we just ask the user to set it themselves, so it's not so bad, but if we want to do this we should maybe consider using a crate, e.g. https://crates.io/crates/set_env.

Reply:
done